test/name_bazaar/smart_contract_tests.cljs

```diff
@@ -71,19 +71,124 @@
+             (let [[[_ {{:keys [:offering]} :args}]]
+                   (alts! [(offering-registry/on-offering-added-once ss
+                                                                     {:node
+                                                                      (namehash
+                                                                       "abc.eth")
+                                                                      :owner (state/my-address 0)})
+                           (timeout 5000)])]
+               (is (not (nil? offering)))
+               (when offering
+                 (is (tx-sent? (<! (registrar/transfer! ss
+                                                        {:ens.record/label "abc" :ens.record/owner offering}
+                                                        {:from (state/my-address 0)}))))
+                 (is (tx-failed?
+                      (<! (buy-now-offering/buy! ss {:offering/address offering} {:value (eth->wei 0.10001)
+                                                                                  :from (state/my-address 1)}))))
+                 (is (tx-failed?
+                      (<! (buy-now-offering/buy! ss {:offering/address offering} {:value (eth->wei 0.09999)
+                                                                                  :from (state/my-address 1)}))))
+                 (is (tx-sent?
+                      (<! (buy-now-offering/buy! ss {:offering/address offering} {:value (eth->wei 0.1)
+                                                                                  :from (state/my-address 1)}))))
+                 (is (tx-failed?
+                      (<! (buy-now-offering/buy! ss {:offering/address offering} {:value (eth->wei 0.1)
+                                                                                  :from (state/my-address 1)}))))
+                 (is (= (state/my-address 1) (last (<! (ens/owner ss {:ens.record/node (namehash
+                                                                                        "abc.eth")}))))))
+               (done)
+               ))
+           (go
+             (is (tx-sent? (<! (registrar/register! ss {:ens.record/label "abc"} {:from (state/my-address 0)}))))
+             (is (tx-sent? (<! (buy-now-offering-factory/create-offering! ss
+                                                                          {:offering/name "abc.eth"
+                                                                           :offering/price (eth->wei 0.1)}
+                                                                          {:from (state/my-address 0)}))))
 
-        ;; TODO more
+             ;;(done)
+             ;; TODO more
 
-        (done)))))
 
+             )))
+  )
+
+(deftest create-auction-offering
+  (async done
+         (let [ss @*server-state*]
+           (go
+             (let [[[_ {{:keys [:offering]} :args}]]
+                   (alts! [(offering-registry/on-offering-added-once ss
```

Comment:
You don't really need to setup event first and only after create offering. you can do register and create-offering first and then to `on-offering-added-once` pass `{:from-block 0}` and it will find the event for sure. It'll be easier to read if code goes in logical sequence.